apps/invoices/src/modules/app-configuration/ui/address-form.tsx

```diff
@@ -1,79 +1,188 @@
+
+const fieldsBlock1: Array<keyof FormSchemaType> = [
+  "companyName",
+  "streetAddress1",
+  "streetAddress2",
+];
+const fieldsBlock2: Array<keyof FormSchemaType> = ["postalCode", "city"];
+const fieldsBlock3: Array<keyof FormSchemaType> = ["cityArea", "country", "countryArea"];
+
+const fieldLabels: Record<keyof FormSchemaType, string> = {
+  countryArea: "Country Area",
+  country: "Country",
+  cityArea: "City Area",
+  streetAddress2: "Street Address 2",
+  streetAddress1: "Street Address 1",
+  companyName: "Company Name",
+  city: "City",
+  postalCode: "Postal Code",
 };
 
 export const AddressForm = (props: Props) => {
-  const { register, handleSubmit } = useForm<SellerShopConfig["address"]>({
-    defaultValues: props.initialData ?? undefined,
+  const { push } = useRouter();
+
+  const channelOverrideConfigQuery = trpcClient.appConfiguration.fetchChannelsOverrides.useQuery();
+
+  const upsertConfigMutation = trpcClient.appConfiguration.upsertChannelOverride.useMutation();
+
+  const { handleSubmit, formState, control, reset } = useForm<SellerAddress>({
+    defaultValues: (channelOverrideConfigQuery.data &&
+      channelOverrideConfigQuery.data[props.channelSlug]) ?? {
+      city: "",
+      postalCode: "",
+      country: "",
+      streetAddress2: "",
+      streetAddress1: "",
+      companyName: "",
+      cityArea: "",
+      countryArea: "",
+    },
+    resolver: zodResolver(FormSchema),
   });
-  const styles = useStyles();
-  const { appBridge } = useAppBridge();
-
-  const CommonFieldProps: TextFieldProps = {
-    className: styles.field,
-    fullWidth: true,
-  };
-
-  const handleChannelNameClick = () => {
-    appBridge?.dispatch(
-      actions.Redirect({
-        to: `/channels/${props.channelID}`,
-      })
-    );
-  };
+
+  const { notifySuccess } = useDashboardNotification();
+
+  useEffect(() => {
+    if (channelOverrideConfigQuery.data) {
+      /**
+       * "Hacky" way to re-render form with new data (from API). Initially it renders empty, because data is not set
+       * Here it re-renders form, but data from API is set, so its provided to defaultValues
+       *
+       * Alternative way will be fetching data in parent and set them in props (and dont render component if data is not set).
+       * This is caused by react hooks that are eager
+       *
+       * TODO it doesnt work with initial render (cmd+r)
+       */
+      reset();
+    }
+  }, [channelOverrideConfigQuery.data, reset, props.channelSlug]);
+
+  if (channelOverrideConfigQuery.isLoading) {
+    return <Text color={"textNeutralSubdued"}>Loading</Text>;
+  }
 
   return (
     <form
       onSubmit={handleSubmit((data, event) => {
-        props.onSubmit(data);
+        return upsertConfigMutation
+          .mutateAsync({
+            address: data,
+            channelSlug: props.channelSlug,
+          })
+          .then(() => {
+            notifySuccess("Success", "Updated channel configuration");
```

Comment:
How will it behave if there's an error?

Is there a reason to not use onSuccess/onError when defining the mutation? 

```
const { mutate } = trpcClient.mjmlConfiguration.getConfiguration.useMutation({
    onSuccess: async (data, variables) => {
      notifySuccess("Configuration saved");
      replace("/configuration");
    },
    onError(error) {
      notifyError(
        "Could not save the configuration"
      );
    },
  });
```

Reply:
done